src/App.tsx

```diff
@@ -24,75 +25,81 @@ import Toast from 'react-native-toast-message';
 import {store} from './reduxApp';
 import {DSN} from './config';
 import {SE} from '@env'; // SE is undefined if no .env file is set
-console.log("> SE", SE)
+import {RootStackParamList} from './navigation';
+console.log('> SE', SE);
 
-const reactNavigationV5Instrumentation = new Sentry.ReactNavigationV5Instrumentation(
-  {
+const reactNavigationInstrumentation =
+  new Sentry.ReactNavigationInstrumentation({
     // How long it will wait for the route change to complete. Default is 1000ms
-    routeChangeTimeoutMs: 500, 
-  },
-);
+    routeChangeTimeoutMs: 500,
+  });
 
 // Get app version from package.json, for fingerprinting
 const packageJson = require('../package.json');
 
 Sentry.init({
   dsn: DSN,
   debug: true,
-  environment: "dev",
+  environment: 'dev',
   beforeSend: (event) => {
-    if (SE === "tda") {
+    if (SE === 'tda') {
       // Make issues unique to the release (app version) for Release Health
-      event.fingerprint = ['{{ default }}', SE, packageJson.version ];
+      event.fingerprint = ['{{ default }}', SE, packageJson.version];
     } else if (SE) {
       // Make issue for the SE
-      event.fingerprint = ['{{ default }}', SE ];
+      event.fingerprint = ['{{ default }}', SE];
     }
-    return event; 
+    return event;
   },
   integrations: [
     new Sentry.ReactNativeTracing({
-      routingInstrumentation: reactNavigationV5Instrumentation,
-      tracingOrigins: ['localhost', /^\//, /^https:\/\//],
-      idleTimeout: 15000 // set to prevent spans in the home screen from cancelling prematurely
+      routingInstrumentation: reactNavigationInstrumentation,
+      tracePropagationTargets: ['localhost', /^\//, /^https:\/\//],
+      idleTimeout: 15000, // set to prevent spans in the home screen from cancelling prematurely
 
       // How to ignore transactions for the "Manual Tracker" screen
-      // beforeNavigate: (context: Sentry.ReactNavigationTransactionContext) => {
-      //   if (context.data.route.name === 'ManualTracker') {
-        //     context.sampled = false;
-        //   }
-        //   return context;
-        // },
+      beforeNavigate: (context: TransactionContext) => {
+        if (context.data?.route.name === 'ManualTracker') {
+          context.sampled = false;
+        }
+        return context;
+      },
     }),
   ],
   tracesSampleRate: 1.0,
   enableAutoSessionTracking: true, // For testing, session close when 5 seconds (instead of the default 30) in the background.
-  sessionTrackingIntervalMillis: 5000, 
+  sessionTrackingIntervalMillis: 5000,
   maxBreadcrumbs: 150, // Extend from the default 100 breadcrumbs.
-  // debug: true
+  attachStacktrace: true,
+  attachScreenshot: true,
+  attachViewHierarchy: true,
 });
 
 Sentry.setTag('se', SE);
 
-const Stack = createStackNavigator();
+const Stack = createStackNavigator<RootStackParamList>();
```

Comment:
is this for an upgraded version of ReactNavigation?

Reply:
I believe it was in the previous version too. It just improves the typings of the props in the screen components. Like [here](https://github.com/sentry-demos/sentry_react_native/pull/35/files#diff-aafd64a88abcae3bdb91b28ad88a0eacd48be83e285bbf6b0c0eb3cfe29d96b2R59-R61). So the `navigator` is not implicitly `any`.